z390test/README.md

```diff
@@ -0,0 +1,243 @@
+# z390test
+
+This folder contains a framework for testing the z390 application.
+
+This is a black box test framework and does not perform unit testing
+of the actual z390 code. It calls the z390 app via a shell command.
+
+The idea is that you can run z390 with input and then check the output 
+from the process.
+
+
+## Setup
+
+Build the z390 Java jar and libraries for use by the tests. From the z390 root:
+
+    ./build.sh
+    win> BUILD.BAT
+
+## Run 
+
+By default the tests will not show the output of the `printOutput` method (see below).
+While developing tests, it is useful to see this output.
+
+    export Z390_PRINT_OUTPUT 1
+    win> SET Z390_PRINT_OUTPUT=1
+
+Run the tests from z390test clone directory
+
+    cd z390test
+    ./gradlew test
+    win> gradlew test
+
+If you use an IDE like IntelliJ or Eclipse, they have support for Gradle test suites.
+
+https://www.jetbrains.com/help/idea/work-with-tests-in-gradle.html
+
+
+    
+## Writing test cases
+
+The following is a basic test case that executes an assembly of a module in the z390 codebase 
+and checks the return code.
+
+```groovy
+import org.junit.jupiter.api.Test
+
+class TestZ390Test extends z390Test {
+
+    var options = ['trace', 'noloadhigh', "SYSMAC(${basePath("mac")})"]
+
+    @Test
+    void test_file_source() {
+        int rc = this.asm(basePath('tests', 'TESTINS1'), *options)
+        this.printOutput()
+        assert rc == 0
+    }
+}
+```
+
+To use the standard framework, your class should extend `z390Test`
+which provides methods for running and capturing the output.
+
+The tests use standard JUnit test structures.
+
+## Methods for testing
+
+The z390Test class embeds a number of methods for interacting with z390 similar
+to the scripts you would use.
+
+### assemble from file
+
+```groovy
+int rc = this.asm(basePath('tests', 'TESTINS1'), *options)
+assert rc == 0
+```
+Parameters:
+* asmFile - The HLASM file without prefix to assemble. Use basePath method to build path relative to z390.
+* args... - 0-n parmeters to pass to the asm program, generally options.
+
+Returns:
+* rc - The result code from the execution
+
+Methods available :
+* asm - Assemble only
+* asml - Assemble and link. Any non zero RC on assemble will exit with RC
+* asmlg - Assemble, link and go. Any non zero RC on assemble and link will exit with RC
+* mz390 - Assemble only
+* cblc - COBOL assemble only
+* cblclg - COBOL assemble, link and go. Any non-zero return will exist with RC
+* lz390 - Link only. Output not cleared prior to run
+* ez390 - Run only. Output not cleared prior to run
```

Comment:
Not sure of your intention here, but mz390 and ez390 are usually not used directly. Instead, use names mac and exec, which correspond to bash/mac and bash/exec. Also, not a problem, but why no cblcl?